core/src/com/moreoptions/prototype/level/Map.java

```diff
@@ -0,0 +1,286 @@
+package com.moreoptions.prototype.level;
+
+import java.util.Random;
+
+/**
+ * Implementation for a randomly generated map
+ *
+ * @author Andreas
+ */
+
+public class Map {
+    private static final int STARTING_ROOM = 1;
+    private static final int NORMAL_ROOM = 2;
+    private static final int BOSS_ROOM = 3;
+    private static final int VENDOR_ROOM = 4;
+    private static final int ITEM_ROOM = 5;
+    private static final int SECRET_ROOM = 6;
+
+    private int[][] map;
+    private int width = 11;
+    private int height = 11;
+
+    private boolean isStartingRoom = true;
+
+
+    private int roomCounter = 0;
+
+    private int maxRooms = 15;
+
+    public Map() {
+        this.map = new int[width + 1][height + 1];
+        createMap();
+        fillMap();
+
+        printMap();
+    }
+
+    /**
+     * gets the number of adjacent rooms for a room.
+     * @param x x-coordinate of the room.
+     * @param y y-coordinate of the room.
+     * @return number of adjacent rooms.
+     */
+    private int getNumberOfAdjacentRooms(int x, int y) {
+        int adjacentRoomsCounter = 0;
+
+        if (!adjacentLeftEmpty(x,y)) {
+            adjacentRoomsCounter++;
+        }
+        if (!adjacentRightEmpty(x,y)) {
+            adjacentRoomsCounter++;
+        }
+        if (!adjacentTopEmpty(x,y)) {
+            adjacentRoomsCounter++;
+        }
+        if (!adjacentBottomEmpty(x,y)) {
+            adjacentRoomsCounter++;
+        }
+
+        return adjacentRoomsCounter;
+    }
+
+    /**
+     * fills the map with a number of rooms.
+     */
+    private void fillMap() {
+        // TODO: better closure for while loop (not maxRooms + X)
+        while (roomCounter <= maxRooms + 4) {
```

Comment:
4 is a constant here, because we always generate 1 treasure, 1 boss, 1 special and 1 vendor.